### PR TITLE
Bugs - C# project assembly proper naming

### PR DESCRIPTION
**Pull request by jicata**

### CODE REVIEW

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/CSharpProjectTestsExecutionStrategy.cs [hunk 1]
```diff
@@ -267,5 +269,26 @@ protected virtual void ExtractTestNames(IEnumerable<TestContext> tests)
...
+            }
+
+            var csProjFullpath = project.FullPath;
+            string projectName = this.ExtractProjectNameFromCsProjfile(csProjFullpath);
+            project.SetProperty("AssemblyName", projectName);
+        }
+
+        private string ExtractProjectNameFromCsProjfile(string csprojFile)
+        {
+            var indexOfLastSlash = csprojFile.LastIndexOf("\\");
```
Comment (Innos) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/CSharpProjectTestsExecutionStrategy.cs]: Use Path.GetFilenameWithoutExtension instead, also there is no need for a separate method, just extract the project name in the EnsureAssemblyNameIsCorrect method, this one seems redundant.

**Comment by Innos** [comment]
Is CSharpPerformanceProjectTests also covered by this change?

**Comment by jicata** [comment]
Yes, CSharpPerformanceProjectTests inherits directly from CSharpProjectTestsExecutionStrategy and overrides only the RunUnitTests method which doesn't interfere with the CorrectProjectReferences method.

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/CSharpAspProjectTestsExecutionStrategy.cs [hunk 2]
```diff
@@ -35,7 +35,7 @@ protected override void CorrectProjectReferences(IEnumerable<TestContext> tests,
             bool nuGetPackageImportsTarget = project.Targets.ContainsKey("EnsureNuGetPackageBuildImports");
             if (nuGetPackageImportsTarget)
             {
-                this.RemoveXmlNodeFromCsProj(project.FullPath,NuGetXmlNodeXPath);
+                this.RemoveXmlNodeFromCsProj(project.FullPath, NuGetXmlNodeXPath);
             }
 
             bool vsToolsImport =
```
Comment (YanaSlavcheva) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/CSharpAspProjectTestsExecutionStrategy.cs]: var

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/CSharpProjectTestsExecutionStrategy.cs [hunk 3]
```diff
@@ -267,5 +269,18 @@ protected virtual void ExtractTestNames(IEnumerable<TestContext> tests)
...
+        protected void EnsureAssemblyNameIsCorrect(Project project)
+        {
+            var assemblyNameProperty = project.AllEvaluatedProperties.FirstOrDefault(x => x.Name == "AssemblyName");
+            if (assemblyNameProperty == null)
+            {
+                throw new ArgumentException("Project file does not contain Assembly Name property!");
+            }
+
+            var csProjFullpath = project.FullPath;
+            string projectName = Path.GetFileNameWithoutExtension(csProjFullpath);
```
Comment (YanaSlavcheva) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/CSharpProjectTestsExecutionStrategy.cs]: var

**Pull request merged**